clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -153,4 +154,16 @@ public class AzureComputeClient extends AzureBaseClient {
     }
     null
   }
+
+  ServiceResponse<Void> destroyServerGroup(String resourceGroupName, String serverGroupName) {
+    try {
+      this.client.getVirtualMachineScaleSetsOperations()?.delete(resourceGroupName, serverGroupName)
+    } catch (CloudException e) {
+      // treat exception as a http 404 return (resource not found)
+      log.warn("ServerGroup: ${e.message} (${serverGroupName} was not found?)")
+    }
```

Comment:
Why are we returning the ServiceResponse? We don't do anything with it when we make the call. Also, the cloud exception should have the status code in the Response member that can tell you if it was a 404 error or not.